src/libponyc/pass/flatten.c

```diff
@@ -279,10 +279,12 @@ ast_result_t pass_flatten(ast_t** astp, pass_opt_t* options)
 
     case TK_FVAR:
```

Comment:
The `TK_FVAR` and `TK_FLET` cases should be completely removed here, or else they'll go through the same checks as `TK_EMBED`, which is wrong.